CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java

```diff
@@ -0,0 +1,235 @@
+        installedFileLocatorMockedStatic.when(() -> InstalledFileLocator.getDefault().locate("modules/ext/icons/drop_down_arrow.png", "au.gov.asd.tac.constellation.utilities", false)).thenReturn(iconFile);
+
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        String expResult = "Text";
+
+        String result = instance.getText();
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test the constructor works with an Icon
+     */
+    @Test
+    public void testConstructor_WithIcon() {
+        final Icon icon = new ImageIcon(COOKIE_ICON_PATH);
+        JDropDownMenu instance = new JDropDownMenu(icon, items);
+        Icon result = instance.getIcon();
+        assertNotNull(result);
+    }
+
+    /**
+     * Test of getText method, of class JDropDownMenu.
+     */
+    @Test
+    public void testGetText() {
+        System.out.println("getText");
+        String text = "Test text";
+        JDropDownMenu instance = new JDropDownMenu(text, items);
+        String result = instance.getText();
+        assertEquals(result, text);
+    }
+
+    /**
+     * Test of setText method, of class JDropDownMenu.
+     */
+    @Test
+    public void testSetText() {
+        System.out.println("setText");
+        String text = "Test text";
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        instance.setText(text);
+        String result = instance.getText();
+        assertEquals(result, text);
+    }
+
+    /**
+     * Test of getIcon method, of class JDropDownMenu.
+     */
+    @Test
+    public void testGetIcon() {
+        System.out.println("getIcon");
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        Icon result = instance.getIcon();
+        assertNotNull(result);
+    }
+
+    /**
+     * Test of setIcon method, of class JDropDownMenu.
+     */
+    @Test
+    public void testSetIcon() {
+        System.out.println("setIcon");
+
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+
+        try {
+            final Icon newIcon1 = new ImageIcon(Utilities.toURI(iconFile).toURL());
+            final Icon newIcon2 = new ImageIcon(DROP_DOWN_ARROW_ICON_PATH);
+            instance.setIcon(newIcon1);
+            Icon result = instance.getIcon();
+            assertEquals(newIcon1, result);
+
+            instance.setIcon(newIcon2);
+            result = instance.getIcon();
+            assertEquals(newIcon2, result);
+        } catch (MalformedURLException ex) {
+            System.out.println("Could not create drop down arrow image");
+        }
+    }
+
+    /**
+     * Test of getItems method, of class JDropDownMenu.
+     */
+    @Test
+    public void testGetItems() {
+        System.out.println("getItems");
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        Set result = instance.getItems();
+        assertEqualsNoOrder(result.toArray(), items.toArray());
```

Comment:
Nice! I'll be using this assert more often, was looking for something like this.